ui/app/src/Views/Shared/SilenceBase.elm

```diff
@@ -63,3 +68,34 @@ matcherButton matcher =
             )
     in
         Utils.Views.labelButton (Just msg) (Utils.List.mstring matcher)
+
+
+editButton : String -> Html Msg
+editButton silenceId =
+    let
+        editUrl =
+            String.join "/" [ "#/silences", silenceId, "edit" ]
+    in
+        a [ class "h-100 btn btn-success rounded-0", href editUrl ]
+            [ span [ class "fa fa-pencil" ] [] ]
+
+
+deleteButton : Silence -> Html Msg
+deleteButton silence =
+    if silence.status.state == "expired" then
+        i [] []
```

Comment:
Maybe change this to `text ""` or make all buttons return `Maybe (Html Msg)`, then you could render them with `List.filterMap identity maybeButtons`